### PR TITLE
Update scala3-library, ... to 3.3.0

### PR DESCRIPTION
**Pull request by pomadchin**
Closes https://github.com/typelevel/cats-tagless/pull/427

### CODE REVIEW

**Comment by joroKr21** [comment]
It doesn't work because we're not publishing macros 😢

**Comment by pomadchin** [comment]
Yes! Give me a bit of time I have an idea, have done smth similar in frameless long time ago :D

File: build.sbt [hunk 1]
```diff
@@ -111,7 +111,8 @@ lazy val macros = crossProject(JVMPlatform, JSPlatform, NativePlatform)
     moduleName := "cats-tagless-macros",
     scalacOptions ~= (_.filterNot(opt => opt.startsWith("-Wunused") || opt.startsWith("-Ywarn-unused"))),
     libraryDependencies += "org.scalacheck" %%% "scalacheck" % scalaCheckVersion % Test,
-    publish / skip := scalaBinaryVersion.value.startsWith("3")
+    publish / skip := scalaBinaryVersion.value.startsWith("3"),
+    mimaPreviousArtifacts := when(scalaBinaryVersion.value.startsWith("2"))(mimaPreviousArtifacts.value.toSeq*).toSet
```
Comment (joroKr21) [inline build.sbt:115]:
```suggestion
    mimaPreviousArtifacts ~= (xs => if (scalaBinaryVersion.value.startsWith("2")) xs else Set.empty)
```

**Review by joroKr21 — APPROVED** [review]
Nice 👍 Btw Scala 3.3.0 includes "Experimental Macro annotations" so we may be able to even have backwards source compatibility in the end.

**Comment by joroKr21** [comment]
@pomadchin sorry, I messed up with git - I wanted to revert my suggestion because it didn't work.
So I did this:
```
git checkout upd/scala-3
git reset --hard HEAD~1
git push -f
```

But I forgot to fetch remote so I had only your initial commit and wiped it out.
And that automatically closed the PR. So I can't even push the original commits to this branch 😂 
So if you still have it locally, you just have to force push and reopen. Sorry for this.

**Pull request closed without merging**